confluent_platform/datadog_checks/confluent_platform/data/metrics.yaml

```diff
@@ -0,0 +1,1276 @@
+# Default metrics collected by this check. You should not have to modify this.
+jmx_metrics:
+  # Kafka Broker Metrics (Yammer Metric Gauge)
+  - include:
+      bean:
+        # Size of the request queue. A congested request queue will not be able to process incoming or outgoing requests
+        - kafka.network:type=RequestChannel,name=RequestQueueSize
```

Comment:
Have you looked into the difference that it makes for JMX fetch to specify metrics this way? Does it try to batch metrics? Does having different domains in the same `include` section have any impact?

If so can we give details in the PR description?

Reply:
Sure I added this to the PR desc.


#### How JMX config works (simplified)

1. Fetch all candidate beans: https://github.com/DataDog/jmxfetch/blob/3485cb33628334d5a26a680826eb645826534d85/src/main/java/org/datadog/jmxfetch/Instance.java#L627-L633

2. Loop through all the beans: https://github.com/DataDog/jmxfetch/blob/3485cb33628334d5a26a680826eb645826534d85/src/main/java/org/datadog/jmxfetch/Instance.java#L474

3. Loop through all the attributes of all beans

4. For each attributes try to match `domain`, `class`, `bean`, `attribute`: https://github.com/DataDog/jmxfetch/blob/3485cb33628334d5a26a680826eb645826534d85/src/main/java/org/datadog/jmxfetch/JmxSimpleAttribute.java#L56-L66 

    If `domain`, `class` or `bean` is a list, the match will succeed if one of the elements in the list matches.

5. If the match succeed, the attribute (correspond to a metric) will be submitted.
 

---

> Does having different domains in the same include section have any impact?

The matching logic will just try to match any of then domains.